File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/11 - For Loop Exercise/02. Half Sum Element .py
num = int(input())

total = 0
max_number = -999999999999999999999999999999

for i in range(1, num + 1):
    in_number = int(input())
    total += in_number
    if in_number > max_number:
        max_number = in_number

total_minus_max = total - max_number

if total_minus_max == max_number:
    print("Yes")
    print(f"Sum = {max_number}")
else:
    print("No")
    print(f"Diff = {abs(total_minus_max - max_number)}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/12. УПРАЖНЕНИЕ ЦИКЛИ/02. Half Sum Element.js
function halfSumElement(params) {
    let count = Number(params.shift());

    let restNumbers = 0;
    let maxNumber = Number.MIN_SAFE_INTEGER;

    for (i = 1; i <= count; i++) {
        number = Number(params.shift());
        restNumbers += number;
        if (number >= maxNumber) {maxNumber = number;}
    }
    restNumbers -= maxNumber

    if (restNumbers == maxNumber) {
        console.log(`Yes`);
        console.log(`Sum = ${maxNumber}`);
    } else {
        console.log(`No`);
        console.log(`Diff = ${Math.abs(maxNumber - restNumbers)}`)
    }
}

halfSumElement([7, 3, 4, 1, 1, 2, 12, 1]);  // Expected Output:
// Yes
// Sum = 12

halfSumElement([4, 6, 1, 2, 3]);  // Expected Output:
// Yes
// Sum = 6

halfSumElement([3, 1, 1, 10]);  // Expected Output:
// No
// Diff = 8


